diff --git a/node_modules/@croct/eslint-plugin/configs/javascript.js b/node_modules/@croct/eslint-plugin/configs/javascript.js
index cd0a260..ccbcb64 100644
--- a/node_modules/@croct/eslint-plugin/configs/javascript.js
+++ b/node_modules/@croct/eslint-plugin/configs/javascript.js
@@ -3,12 +3,10 @@ Object.defineProperty(exports, "__esModule", { value: true });
 exports.javascript = void 0;
 exports.javascript = {
     extends: [
-        'plugin:jest/recommended',
         'airbnb-base',
         'plugin:eslint-comments/recommended',
     ],
     plugins: [
-        'jest',
         'import',
         'no-smart-quotes',
         'import-newlines',
@@ -40,26 +38,6 @@ exports.javascript = {
             'always-multiline',
         ],
         'no-undef-init': 'error',
-        'jest/consistent-test-it': [
-            'error',
-            {
-                fn: 'it',
-            },
-        ],
-        'jest/no-large-snapshots': 'off',
-        'jest/prefer-expect-resolves': 'error',
-        'jest/prefer-lowercase-title': [
-            'error',
-            {
-                ignore: ['describe'],
-            },
-        ],
-        'jest/prefer-spy-on': 'error',
-        'jest/require-top-level-describe': 'error',
-        'jest/prefer-to-contain': 'error',
-        'jest/prefer-hooks-on-top': 'error',
-        'jest/prefer-equality-matcher': 'error',
-        'jest/no-test-return-statement': 'error',
         'function-call-argument-newline': [
             'error',
             'consistent',
@@ -246,20 +224,4 @@ exports.javascript = {
             'prefer-double',
         ],
     },
-    overrides: [
-        {
-            files: [
-                'src/**/*.test.js',
-                'test/**/*.js',
-            ],
-            extends: ['plugin:jest/recommended'],
-            plugins: ['jest'],
-            rules: {
-                'no-new-object': 'off',
-            },
-            env: {
-                jest: true,
-            },
-        },
-    ],
 };
diff --git a/node_modules/@croct/eslint-plugin/configs/typescript.js b/node_modules/@croct/eslint-plugin/configs/typescript.js
index 715c095..bd67846 100644
--- a/node_modules/@croct/eslint-plugin/configs/typescript.js
+++ b/node_modules/@croct/eslint-plugin/configs/typescript.js
@@ -9,7 +9,7 @@ exports.typescript = {
     ],
     overrides: [
         {
-            files: ['**/*.ts', '**/*.tsx'],
+            files: ['**/*.ts', '**/*.mts', '**/*.cts', '**/*.tsx'],
             extends: ['plugin:@typescript-eslint/recommended'],
             parser: '@typescript-eslint/parser',
             rules: {
@@ -81,19 +81,5 @@ exports.typescript = {
                 '@typescript-eslint/no-namespace': 'off',
             },
         },
-        {
-            files: [
-                'src/**/*.test.ts',
-                'test/**/*.ts',
-            ],
-            extends: ['plugin:jest/recommended'],
-            plugins: ['jest'],
-            rules: {
-                'no-new-object': 'off',
-            },
-            env: {
-                jest: true,
-            },
-        },
     ],
 };
